### PR TITLE
Adding nccmp (a tool for comparison of NetCDF files)

### PR DESCRIPTION
**Pull request by zklaus**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/nccmp/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+source:
+  url: https://gitlab.com/remikz/{{ name }}/-/archive/{{ version }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 4e86d2ea788191a36b29ce09a50dcc3711abd100ff36a1d4e2d0d19f4de75fbd
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - cmake
```
Comment (scopatz) [inline recipes/nccmp/meta.yaml]: This should also depend on make

**Comment by zklaus** [comment]
I think this is ready. @scopatz, are you interested to become co-maintainer?

**Comment by zklaus** [comment]
@ocefpaf, maybe this falls into your area of interest as well?

**Comment by scopatz** [comment]
> I think this is ready. @scopatz, are you interested to become co-maintainer? 

That's Ok, thanks!

File: recipes/nccmp/build.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+mkdir build && cd build
+cmake \
+    -DBUILD_TESTS=ON \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_EXE_LINKER_FLAGS="-L$PREFIX/lib -lhdf5 -lhdf5_hl -lcurl -lnetcdf" \
+    -DCMAKE_INSTALL_PREFIX=$PREFIX \
+    -DCMAKE_VERBOSE_MAKEFILE=OFF \
+    ..
+
+make -j
```
Comment (scopatz) [inline recipes/nccmp/build.sh]: should be `make -j ${CPU_COUNT}`

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**